blockchain/blockchain.go

```diff
@@ -1005,7 +1005,7 @@ func (b *Blockchain) writeBody(batchWriter *storage.BatchWriter, block *types.Bl
 
 	// Write txn lookups (txHash -> block)
 	for _, txn := range block.Transactions {
-		batchWriter.PutTxLookup(txn.Hash, block.Hash())
+		batchWriter.PutTxLookup(txn.GetHash(), block.Hash())
```

Comment:
Why not? I think it's worth it

Reply:
I was thinking because, most of the time it is from a single goroutine and we face multiple goroutines in unit test execution since we are executing them in parallel. So, although performance overhead is probably negligible, we may acquire a read lock even when we do not need it.

Reply:
I assume that this change comes from a fix in CI and as performance is negligible and change is future proof, for me it's worth it

Reply:
> I assume that this change comes from a fix in CI.

In fact, this change was not initiated (https://github.com/0xPolygon/polygon-edge/pull/1925/commits/760cfe5f2fba4faaebb9f9450decf44ee6c34b78), because CI was complaining, but I thought that maybe we should access it in a thread-safe manner, just to be future-proof as you said.
So eventually I don't mind dropping the commit.